File: backend/src/main/java/com/awais/hr/module/communication/service/InternalCommunicationService.java
package com.awais.hr.module.communication.service;

import java.util.List;
import java.util.Map;

public interface InternalCommunicationService {
    List<Map<String, Object>> getFeedPosts();
    Map<String, Object> createFeedPost(Map<String, Object> body);
    List<Map<String, Object>> getPolls();
    Map<String, Object> createPoll(Map<String, Object> body);
}

File: frontend/src/services/internalCommunicationService.js
import apiClient from './api';

export const getFeedPosts = () => apiClient.get('/suite/internal-communication/posts');
export const createFeedPost = (dto) => apiClient.post('/suite/internal-communication/posts', dto);
export const getPolls = () => apiClient.get('/suite/internal-communication/polls');
export const createPoll = (dto) => apiClient.post('/suite/internal-communication/polls', dto);
